File: client/java/src/avt/AnimalDan.java
package avt;

public class AnimalDan extends Animal {
   public int captainID = 0;
   public byte indexFood;

   public AnimalDan(int var1, int var2, int var3, byte var4) {
      super(0, 0, var3, var4);
   }

   public void update() {
      super.update();
   }

   public final void setAngleAndDis() {
      super.setAngleAndDis();
      if (!super.isEat && super.IDDB == this.captainID && super.distant > 150) {
         super.distant = 150;
      }

   }

   public void setInit() {
   }

   public Point getPosEat() {
      return (Point)FarmScr.listFood[this.indexFood].elementAt(CRes.rnd(FarmScr.listFood[this.indexFood].size()));
   }

   public final void updatePos() {
      if (!super.isEat && this.captainID == super.IDDB) {
         this.setPos();
      } else {
         AvPosition var1 = new AvPosition();
         if (super.isEat && FarmScr.listFood[this.indexFood].size() > 0) {
            Point var5;
            if ((var5 = this.getPosEat()) != null) {
               var1.x = var5.x;
               var1.y = var5.y;
               super.v = 2;
               super.posNext = var1;
            } else {
               this.setPos();
            }
         } else {
            int var2 = LoadMap.playerLists.size();

            for(int var3 = 0; var3 < var2; ++var3) {
               Base var4;
               if ((var4 = (Base)LoadMap.playerLists.elementAt(var3)) instanceof AnimalDan && var4.IDDB == this.captainID) {
                  var1 = new AvPosition(var4.x, var4.y);
                  break;
               }
            }

            if (this.indexFood != 1 && !LoadMap.isTrans(super.x, super.y)) {
               this.setPos();
            } else {
               this.setFollowPos(var1);
            }
         }
      }

   }

   public void setFollowPos(AvPosition var1) {
   }

   public final void reset() {
      int var1 = FarmScr.listFood[this.indexFood].size();
      if (super.hunger && super.isEat && var1 > 0) {
         for(int var2 = 0; var2 < var1; ++var2) {
            Point var3;
            if (CRes.abs((var3 = (Point)FarmScr.listFood[this.indexFood].elementAt(var2)).x - super.x) <= 2 && CRes.abs(var3.y - super.y) <= 2) {
               FarmScr.listFood[this.indexFood].removeElement(var3);
               LoadMap.dynamicLists.removeElement(var3);
               super.hunger = false;
               super.isEat = false;
               super.v = 1;
               FarmScr.gI();
               FarmScr.doEat(var3.itemID, super.IDDB);
               break;
            }
         }
      }

      super.reset();
      super.cycle = 100 - (this.captainID != super.IDDB ? this.indexFood * CRes.rnd(70) : 0);
   }

   public final void updateEat() {
      if (FarmScr.listFood[this.indexFood].size() == 0) {
         super.isEat = false;
      } else if (super.hunger && !super.isEat) {
         super.isEat = true;
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/AnimalDan.cs
public class AnimalDan : Animal
{
	public int captainID;

	public sbyte indexFood;

	public AnimalDan(int x, int y, int id, sbyte species)
		: base(x, y, id, species)
	{
	}

	public override void update()
	{
		base.update();
	}

	public override void setAngleAndDis()
	{
		base.setAngleAndDis();
		if (!isEat && IDDB == captainID && distant > 150)
		{
			distant = 150;
		}
	}

	public virtual Point getPosEat()
	{
		return (Point)FarmScr.listFood[indexFood].elementAt(CRes.rnd(FarmScr.listFood[indexFood].size()));
	}

	public override void updatePos()
	{
		if (!isEat && captainID == IDDB)
		{
			setPos();
			return;
		}
		AvPosition avPosition = new AvPosition();
		if (isEat && FarmScr.listFood[indexFood].size() > 0)
		{
			Point posEat = getPosEat();
			if (posEat != null)
			{
				avPosition.x = posEat.x;
				avPosition.y = posEat.y;
				v = 2;
				setPosNext(avPosition);
			}
			else
			{
				setPos();
			}
			return;
		}
		int num = LoadMap.playerLists.size();
		for (int i = 0; i < num; i++)
		{
			Base @base = (Base)LoadMap.playerLists.elementAt(i);
			if (@base is AnimalDan && @base.IDDB == captainID)
			{
				avPosition = new AvPosition(@base.x, @base.y);
				break;
			}
		}
		if (indexFood != 1 && !LoadMap.isTrans(x, y))
		{
			setPos();
		}
		else
		{
			setFollowPos(avPosition);
		}
	}

	public virtual void setFollowPos(AvPosition pos)
	{
	}

	public override void reset()
	{
		int num = FarmScr.listFood[indexFood].size();
		if (hunger && isEat && num > 0)
		{
			for (int i = 0; i < num; i++)
			{
				Point point = (Point)FarmScr.listFood[indexFood].elementAt(i);
				if (CRes.abs(point.x - x) <= 2 && CRes.abs(point.y - y) <= 2)
				{
					FarmScr.listFood[indexFood].removeElement(point);
					LoadMap.dynamicLists.removeElement(point);
					hunger = false;
					isEat = false;
					v = 1;
					FarmScr.gI().doEat(point.itemID, IDDB);
					break;
				}
			}
		}
		base.reset();
		cycle = 100 - ((captainID != IDDB) ? (indexFood * CRes.rnd(70)) : 0);
	}

	public override void updateEat()
	{
		if (FarmScr.listFood[indexFood].size() == 0)
		{
			isEat = false;
		}
		else if (hunger && !isEat)
		{
			isEat = true;
		}
	}
}
